### PR TITLE
[sonic-pit] Add PIT(Platform Integration Test) feature, second part, …

### PR DESCRIPTION
**Pull request by clarklee-guizhao**
…add 6 test cases.

Signed-off-by: Li Hua <guizhao.lh@alibaba-inc.com>

#### Why I did it
sonic-pit is originated from S3IP(Simplify Switch System Integration Program) PIT(Platform Integration Test). It's designed to verify switch devices in the process of development, manufacturing and delivering stages. It focus on 2 roles: ODM and user. ODM may use PIT in developing and manufaturing stages, user may use PIT in delivery stage for acceptance test. It aims to provide an verify system for siwtch devices' hardware, drivers and firmware. More information, see HLD: https://github.com/clarklee-guizhao/SONiC/blob/pit/doc/pit/Platform_Integration_Test_high_level_design.md

#### How I did it
Porting sonic-pit from S3IP-PIT, make the original code clean enough, add PIT source as sonic-pit, add PIT running environment as docker-pit.
The whole PIT system is divided into several PR, this is the second PR with 6 test cases only.

#### How to verify it
First of all, this PR should be merged with the first PR(which has the PIT framework). After a sucessful build, copy sonic_pit-1.0-py3-none-any.whl to your device running sonic, install it with pip install sonic_pit-1.0-py3-none-any.whl, then run '/usr/local/pit-sysdiag/src/sysdiag -d' to start test.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

### CODE REVIEW

**Comment by clarklee-guizhao** [comment]
Add PIT HLD document. #1014

**Comment by clarklee-guizhao** [comment]
https://github.com/sonic-net/SONiC/pull/1014

**Comment by clarklee-guizhao** [comment]
@lguohan Hi Guohan, please help review this PR, thanks. This is the second part of sonic-pit, which only inlucde some test case logic.

File: src/sonic-pit/pit-sysdiag/src/ssd_tc.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
+import os
+import re
+import subprocess
+from test_case import TestCaseCommon
+from errcode import *
+import traceback
+
+
+def run_command(cmd):
```
Comment (keboliu) [inline src/sonic-pit/pit-sysdiag/src/ssd_tc.py:9]: there is already a similar function(if not the same) in PR1, which shall be re-used.

File: src/sonic-pit/pit-sysdiag/src/memory_tc.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        if status != 0 or len(out) <= 0:
+            self.fail_reason.append("dmidecode exec failed.")
+            ret = E.EMEM4001
+        else:
+            for item in out.splitlines():
+                self.logger.log_info(item)
+                if ":" in item:
+                    key = item.split(":")[0].strip()
+                    value = item.split(":")[1].strip()
+                    #self.logger.log_info("%-30s: %s" % (key, value), True)
```
Comment (keboliu) [inline src/sonic-pit/pit-sysdiag/src/memory_tc.py:44]: Please remove if not needed.

File: src/sonic-pit/pit-sysdiag/src/ssd_tc.py (hunk 1 above)
Comment (clarklee-guizhao) [reply src/sonic-pit/pit-sysdiag/src/ssd_tc.py]: Removed in latest commit.

File: src/sonic-pit/pit-sysdiag/src/memory_tc.py (hunk 2 above)
Comment (clarklee-guizhao) [reply src/sonic-pit/pit-sysdiag/src/memory_tc.py]: Removed in latest commit

**Review by keboliu — APPROVED** [review]

**Comment by zhangyanzhao** [comment]
@lguohan  can you please check if this can be merged? Thanks.

**Pull request merged**